### PR TITLE
feat(regulations): update regulations deps and types

### PR DESCRIPTION
**Pull request by Valur**
## What

Use open source @island.is/regulations-tools for all regulations related data

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by thordurhhh — APPROVED** [review]
Nice, LGTM. 👍

File: apps/web/components/Regulations/useRegulationIndexer.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import { HTMLText } from '@hugsmidjan/regulations-editor/types'
-import { Regulation, RegulationMaybeDiff } from '@island.is/regulations/web'
+import { HTMLText } from '@island.is/regulations-tools/types'
+import { Regulation, RegulationMaybeDiff } from '@island.is/regulations'
```
Comment (maranomynet) [inline apps/web/components/Regulations/useRegulationIndexer.ts:2]:
```suggestion
import {
  Regulation,
  RegulationMaybeDiff,
  HTMLText
} from '@island.is/regulations'
```

**Review by maranomynet — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -65,8 +65,8 @@
     "@formatjs/intl-locale": "2.4.33",
     "@formatjs/intl-numberformat": "7.1.5",
     "@hookform/error-message": "0.0.5",
-    "@hugsmidjan/regulations-editor": "0.4.8",
     "@island.is/login": "1.2.1",
+    "@island.is/regulations-tools": "0.7.5",
```
Comment (eirikurn) [inline package.json:69]: 👏

File: yarn.lock [hunk 3]
```diff
@@ -4871,23 +4871,17 @@
...
+  integrity sha512-DxY6bqcXVnhYFQVAAUmrUYR7JE/199XW/hyuoz4/fsyr5CJGlhZGOuoZp9DoDKot3oU99zxLTP4uCCfZDzbR1Q==
   dependencies:
-    "@hugsmidjan/qj" "^4.7.0"
-    "@tinymce/tinymce-react" "^3.8.4"
-    cleanup-prettier "npm:prettier@2.2.0"
-    htmldiff-js "^1.0.5"
-    jsdom "^16.4.0"
-    tinymce "^5.6.2"
-    translate.js "^1.3.2"
+    "@hugsmidjan/qj" "^4.1.1"
```
Comment (eirikurn) [inline yarn.lock:4884]: Hver er staðan á að draga þetta líka inn eða open sourca hjá ykkur með permissive license?

**Review by eirikurn — APPROVED** [review]
Yay!

File: yarn.lock (hunk 3 above)
Comment (Valur) [reply yarn.lock:4884]: > What's the status on getting rid of this, or open sourcing it on your side with a permissive license?

Both of these packages are open source and released under MIT license:
https://github.com/hugsmidjan-is/react-utils
https://github.com/hugsmidjan/qj

**Review by maranomynet — APPROVED** [review]

File: yarn.lock (hunk 3 above)
Comment (eirikurn) [reply yarn.lock:4884]: Sweet! 👏

**Pull request merged**